### PR TITLE
Add Makefile to simplify upstream releases.

### PR DESCRIPTION
**Pull request by mattclay**
##### SUMMARY

Add Makefile to simplify upstream releases.

##### ISSUE TYPE

Feature Pull Request

##### COMPONENT NAME

packaging/release/Makefile

##### ANSIBLE VERSION

```
ansible 2.8.0.dev0 (release-makefile 688d6bd394) last updated 2018/10/31 19:59:48 (GMT -700)
  config file = None
  configured module search path = [u'/Users/mclay/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/mclay/code/mattclay/ansible/lib/ansible
  executable location = /Users/mclay/code/mattclay/ansible/bin/ansible
  python version = 2.7.14 (default, Mar 22 2018, 11:39:16) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]
```

### CODE REVIEW

**Comment by mattclay** [comment]
cc @nitzmahone

**Comment by mattclay** [comment]
This should be able to be backported to stable-2.7 and stable-2.6 to simplify the release process.

It won't work for stable-2.5 since that branch is using reno.

**Comment by mattclay** [comment]
cc @abadger

**Comment by maxamillion** [comment]
LGTM 👍

File: packaging/release/Makefile [hunk 1]
```diff
@@ -0,0 +1,60 @@
+version ?= $(shell python versionhelper/version_helper.py --raw)
+
+.PHONY: all
+all:
+	@echo "USAGE:"
+	@echo
+	@echo "make release version={version} # current version is '${version}'"
+	@echo "make publish"
+	@echo
+	@echo "NOTE: Make sure to source hacking/env-setup before running these targets."
```
Comment (gundalow) [inline packaging/release/Makefile:10]: How bad is it if someone doesn't do this?

Do we need to detect/fail gracefully if this hasn't been done, or as we assuming that people are following the documented process so this should be OK?

**Review by gundalow — COMMENTED** [review]
+1 to automation

File: packaging/release/Makefile (hunk 1 above)
Comment (nitzmahone) [reply packaging/release/Makefile:10]: The failure mode depends on what they have laying around. If there's an `ansible-doc` already on the path, the generated "new plugins" metadata will be incorrect (based on whatever version it is, likely empty in most cases). If they don't have one, it'll just fail when it tries to run it.

The only way to actually detect it that I can think of (and still not 100% bulletproof, but pretty good) would be to parse the version from `ansible-doc --version` and ensure that it matches what's being requested.

**Review by nitzmahone — APPROVED** [review]

File: packaging/release/Makefile (hunk 1 above)
Comment (mattclay) [reply packaging/release/Makefile:10]: I think it's also important to note that this isn't a new issue. Our current manual instructions also require sourcing of `hacking/env-setup`, with the same consequences if you don't.

**Pull request merged**